File: src/ts/src/adjust/IAdjust.ts
import { IPlugin } from "../core";
import { AdjustConfig } from "./AdjustConfig";

export interface IAdjust extends IPlugin {
    initialize(config: AdjustConfig): void;
    setEnabled(enabled: boolean): void;
    getAdvertisingIdentifier(): Promise<string>;
    getDeviceIdentifier(): string;
    setPushToken(token: string): void;
    trackEvent(token: string): void;
}

File: src/unity/Runtime/Adjust/IAdjust.cs
using System.Threading.Tasks;

namespace EE {
    public interface IAdjust : IPlugin {
        void Initialize(AdjustConfig config);
        void SetEnabled(bool enabled);
        Task<string> GetAdvertisingIdentifier();
        string GetDeviceIdentifier();
        void SetPushToken(string token);
        void TrackEvent(string token);
    }
}